### ISSUE TITLE
redundantSelf not working in file that contains switch with nested conditional compilation

### ISSUE THREAD

**Issue by fphilipe** [nicklockwood/SwiftFormat#404]
Using the latest version 0.40.4 and the default options, the following code

```swift
struct Foo {
    func bar() {
        self.method(self.value)

        switch something {
        #if FLAG
            case .whaaat:
                break
        #endif
        default:
            break
        }
    }
}
```

is expected to be formatted as

```swift
struct Foo {
    func bar() {
        method(value)

        switch something {
        #if FLAG
            case .whaaat:
                break
        #endif
        default:
            break
        }
    }
}
```

<details><summary>(expected diff)</summary>

```diff
 struct Foo {
     func bar() {
-        self.method(self.value)
+        method(value)
 
         switch something {
         #if FLAG
             case .whaaat:
                 break
         #endif
         default:
             break
         }
     }
 }
```

</details>

but the **`self` prefixes are not being stripped**. Removing the conditional compilation within the `switch` statement results in the expected removal of redundant `self` prefixes.

From my non-exhaustive tests this seems to be isolated to conditional compilation within `switch` and to the *redundantSelf* rule.

**Comment by nicklockwood** [comment]
@fphilipe thanks for reporting it, I'll investigate.

**Comment by nicklockwood** [comment]
@fphilipe fixed in 0.40.5
